### PR TITLE
added precision, recall, f1_score

### PR DESCRIPTION
**Pull request by mthorrell**
would like to have some type checking since precision, recall and f1 depend on 0 and 1 predictions, but basics should be here. Referencing issue #19.

### CODE REVIEW

File: R/binary_classification.R [hunk 1]
```diff
@@ -76,3 +76,44 @@ ll <- function(actual, predicted) {
 logLoss <- function(actual, predicted) {
     return(mean(ll(actual, predicted)))
 }
+
+
+
+#' Precision
+#' 
+#' \code{precision} computes proportion of predicted 1's that are actual 1's
```
Comment (mfrasco) [inline R/binary_classification.R]: Can you change this to, "computes proportion of observations predicted to be in the positive class (i.e. the element in \code{predicted} equals 1) that actually belong to the positive class (i.e. the element in \code{actual} equals 1)?

Thanks

File: R/binary_classification.R [hunk 2]
```diff
@@ -76,3 +76,44 @@ ll <- function(actual, predicted) {
...
+#' @export
```
Comment (mfrasco) [inline R/binary_classification.R]: Can you add `#' @seealso \code{\link{recall}} \code{\link{fbeta_score}}`

Comment (mfrasco) [inline R/binary_classification.R]: Can you also add `#' @inheritParams params_binary`

File: R/binary_classification.R [hunk 3]
```diff
@@ -76,3 +76,44 @@ ll <- function(actual, predicted) {
...
+#' @examples 
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 1, 1, 1, 1, 1)
+#' precision(actual, predicted)
+precision <- function(actual, predicted) {
+  return(mean(actual[predicted == 1]))
```
Comment (mfrasco) [inline R/binary_classification.R]: Can you use four spaces as the indentation?

File: R/binary_classification.R [hunk 4]
```diff
@@ -76,3 +76,44 @@ ll <- function(actual, predicted) {
...
+}
+
+#' Recall
+#' 
+#' \code{recall} computes proportion of actual 1's that are predicted 1's
+#' @export
+#' @examples 
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' recall(actual, predicted)
```
Comment (mfrasco) [inline R/binary_classification.R]: Can you add all of the same documentation improvements that were listed for `precision`?

File: R/binary_classification.R [hunk 5]
```diff
@@ -76,3 +76,44 @@ ll <- function(actual, predicted) {
...
+recall <- function(actual, predicted) {
+  return(mean(predicted[actual == 1]))
```
Comment (mfrasco) [inline R/binary_classification.R]: four spaces

File: R/binary_classification.R [hunk 6]
```diff
@@ -76,3 +76,44 @@ ll <- function(actual, predicted) {
...
+}
+
+#' F1 score
```
Comment (mfrasco) [inline R/binary_classification.R]: This should be F-beta Score

File: R/binary_classification.R [hunk 7]
```diff
@@ -76,3 +76,44 @@ ll <- function(actual, predicted) {
...
+#' 
+#' \code{f1_score} computes the f1 score
+#' @export
+#' @seealso \code{\link{precision}}, \code{\link{recall}}
+#' @examples 
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' recall(actual, predicted)
+f1_score <- function(actual, predicted, beta = 1) {
```
Comment (mfrasco) [inline R/binary_classification.R]: `fbeta_score`

File: R/binary_classification.R [hunk 8]
```diff
@@ -76,3 +76,44 @@ ll <- function(actual, predicted) {
...
+
+#' F1 score
+#' 
+#' \code{f1_score} computes the f1 score
+#' @export
+#' @seealso \code{\link{precision}}, \code{\link{recall}}
+#' @examples 
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' recall(actual, predicted)
```
Comment (mfrasco) [inline R/binary_classification.R]: Same documentation improvements

File: R/binary_classification.R [hunk 9]
```diff
@@ -76,3 +76,44 @@ ll <- function(actual, predicted) {
...
+recall <- function(actual, predicted) {
+  return(mean(predicted[actual == 1]))
+}
+
+#' F1 score
+#' 
+#' \code{f1_score} computes the f1 score
+#' @export
+#' @seealso \code{\link{precision}}, \code{\link{recall}}
```
Comment (mfrasco) [inline R/binary_classification.R]: You don't need the comma here

File: R/binary_classification.R [hunk 10]
```diff
@@ -76,3 +76,44 @@ ll <- function(actual, predicted) {
...
+#' @examples 
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' recall(actual, predicted)
+f1_score <- function(actual, predicted, beta = 1) {
+  prec = precision(actual, predicted)
+  rec = recall(actual, predicted)
```
Comment (mfrasco) [inline R/binary_classification.R]: Can you use the assignment operator `<-` and use four spaces.

File: tests/testthat/test-binary_classification.R [hunk 11]
```diff
@@ -21,3 +21,21 @@ test_that('mean los loss is calculated correctly', {
...
+  expect_equal(recall(c(1,1,0,0),c(1,1,0,0)), 1)
+  expect_equal(recall(c(0,0,1,1),c(1,1,0,0)), 0)
+  expect_equal(recall(c(1,1,1,1),c(1,0,0,1)), 1/2)
+})
+
+test_that('f1 score is calculated correctly',{
+  expect_equal(f1_score(c(1,1,0,0),c(1,1,0,0)), 1)
+  expect_equal(f1_score(c(0,0,1,1),c(1,1,1,0)), 2/5)
+  expect_equal(f1_score(c(1,1,1,1),c(1,0,0,1)), 2/3)
+})
```
Comment (mfrasco) [inline tests/testthat/test-binary_classification.R]: Thanks for the unit tests!

**Comment by mthorrell** [comment]
thanks for the comments. Let me know your feedback on the F-beta score descriptions I made if you'd like that changed or if you want it to be more specific.

If this looks good, still to be done is to update the TeX equations on the main page.

**Comment by mfrasco** [comment]
I think that this looks great. I like the descriptions of the `fbeta_score` function. In your original comment on this PR, you mentioned that you would like to do type checking on the inputs. I'm going to defer that work to a different PR, as the type checking needs to be made throughout the repo.

If you could add the equations to the README, I'll approve the PR.

**Comment by mthorrell** [comment]
Sorry for the delay. I think this does it.

**Comment by dimagor** [comment]
Really excited to see these changes make it into this awesome package. Any thoughts on when this PR will be merged and ideally up on CRAN?

**Comment by mfrasco** [comment]
@mthorrell I must have missed the email notification that alerted me to the fact that you pushed changes a month ago! I'm sorry. This all looks good. I'll merge this now. @dimagor I can push to CRAN this weekend.

**Comment by dimagor** [comment]
@mfrasco thank you for merging this. Any luck with the CRAN update?

**Comment by mfrasco** [comment]
@dimagor On CRAN now. Sorry for the delay. I missed the email from CRAN that described the changes that I needed to make in order to get it approved.

**Comment by dimagor** [comment]
Fantastic! Thanks a lot!!

**Pull request merged**